miden-lib/asm/eoa/basic.masm

```diff
@@ -1,32 +1,47 @@
 use.miden::sat::account
 use.miden::sat::tx
+use.std::crypto::dsa::rpo_falcon512
 
-#! Authenticate a transaction using Falcon
-#! WIP - FALCON VERIFICATION IS MISSING, NOT SURE IF HASHING IN CORRECT ORDER
+# CONSTANTS
+# =================================================================================================
+
+# Slot in account storage at which the public key is stored.
+const.PUBLIC_KEY_SLOT=0
+
+#! Authenticate a transaction using Falcon 
 #! Stack: []
 #! Output: []
 #!
-#! - ASSET is the non-fungible asset of interest.
-#! - tag is the tag to be included in the note.
-#! - RECIPIENT is the recipient of the note.
 export.auth_tx_rpo_falcon512
-    # update the nonce
-    push.1
-    exec.account::incr_nonce
-
-    # get commitments to consumed and created notes, new nonce, and ID
-    #exec.tx::get_output_notes_hash
-    #exec.tx::get_input_notes_hash
-    #exec.account::get_nonce push.0.0.0
-    #exec.account::get_id push.0.0.0
-
-    # compute the message to be signed
-    # M = h(output_notes_hash, h(input_notes_hash, h(0, 0, 0, id, 0, 0, 0, nonce)))
-    #hmerge hmerge hmerge
-
-    # get public key from account storage (assuming it is stored at index 0) and verify signature
-    #push.0
-    #exec.account::get_item
-    # [PUB_KEY, M]
-    # TODO: sign the message using FALCON
+    # Get commitments to consumed and created notes, current nonce, and ID
```

Comment:
optional nit: it may be preferable to split this comment up and comment the lines below according to the entity being fetched such that each operation has a comment.